### PR TITLE
Rails Pathway: Provide descriptive link text. 

### PR DESCRIPTION
**Pull request by thatsthat**
## Because

Links text need to be update to comply to the guide on accessible, descriptive link text

## This PR

* Updated lesson: Turbo
* Updated lesson: Stimulus
* Updated lesson: Mailers
* Updated Project: Sending Confirmation Emails

## Issue

Related to #28290

## Additional Information

Updated according to [spreasdheet] (https://docs.google.com/spreadsheets/d/1rM9tjPIFZbKoMfd6OlLnjOhddBe2bivllY8_Bg5aGqY/edit?usp=sharing)

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project curriculum contributing guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x ] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

**Comment by MaoShizhong** [comment]
A heads up that we have instructions and scripts for local linting in the [contributing guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md) for this repo (CONTRIBUTING.md).

Before you use the scripts though, there are a couple of manual changes you need to make (else the scripts will apply incorrect fixes in a few places):
1. `mailers.md` line 58, manually change the `*` to `-`. Looks like you missed this when manually changing but if you don't manually fix it, it'll treat it as a new list and surround it with blank lines rather than changing it to a `-`.
2. `mailers.md` line 79, The markdown link starts with `]` instead of `[` by mistake.
3. `stimulus_js.md` line 301, you need to add the backslashes back in so the markdown doesn't treat the text as inline HTML.

After you've fixed the above, the appropriate fix scripts should fix the majority of issues.
You don't have to fix any errors that can't be handled via the scripts, as they'd be context-dependent things to handle.

**Comment by thatsthat** [comment]
Hi, I am not sure what is wrong with the remaining linter issue, the [offending link](https://www.theodinproject.com/lessons/ruby-on-rails-stimulus#selecting-targeting-elements) seems to be correct:

Error: ruby_on_rails/rails_sprinkles/stimulus_js.md:310:3 MD051/link-fragments Link fragments should be valid [Context: "[How do you select a DOM element?](#selecting-targeting-elements)"] https://github.com/DavidAnson/markdownlint/blob/v0.32.1/doc/md051.md

File: ruby_on_rails/mailers_advanced_topics/mailers.md [hunk 1]
```diff
@@ -46,37 +48,42 @@ You'll get a chance to play with mailers in the projects.
...
-* <a class='knowledge-check-link' href='#brief-overview'>How do you create a new mailer from the command line?</a>
-* <a class='knowledge-check-link' href='#brief-overview'>Where do you store the actual email in your application?</a>
-* <a class='knowledge-check-link' href='#brief-overview'>How do you send an email directly from the Rails console?</a>
-* <a class='knowledge-check-link' href='#callbacks'>How do you use callbacks with mailers?</a>
-* <a class='knowledge-check-link' href='#email-wisdom'>How do you write links in mailer views?</a>
-* <a class='knowledge-check-link' href='#email-wisdom'>How do you style an HTML email?</a>
+The following questions are an opportunity to reflect on key topics in this lesson. If you can't answer a question, click on it to review the material, but keep in mind you are not expected to memorize or master this knowledge.
+
+- [How do you create a new mailer from the command line?](#brief-overview)
+- ]Where do you store the actual email in your application?](#brief-overview)
```
Comment (MaoShizhong) [inline ruby_on_rails/mailers_advanced_topics/mailers.md:82]:
```suggestion
- [Where do you store the actual email in your application?](#brief-overview)
```

File: ruby_on_rails/mailers_advanced_topics/mailers.md [hunk 2]
```diff
@@ -46,37 +48,42 @@ You'll get a chance to play with mailers in the projects.
...
-* Attaching images to emails (like using logos in the HTML version) can be a bit of a pain.  See the reading.
+- Email is SLOW! It can take 1-2 seconds per email to send, so don't make your main application do it when you're serving a whole bunch of them because then anyone trying to access it will be shut out.
+- Make sure you use full URLs in any links in your mailer (so `_url` not `_path` helper methods), since the user will be opening the email and clicking the link at an external source.  In your `config/environments/production.rb` file you'll want to make sure to specify your website's host name using something like `config.action_mailer.default_url_options = { :host => 'yourapp.com' }`.  If it's not set, you may get an error message about your host or your links may look funny.
+- The `#deliver!` method will throw an error on delivery failure whereas `#deliver` will return false (failing silently).
+- When styling your email HTML, you won't have access to any stylesheets so you'll need to do all the styling either inline or using `<style>` tags.
+
```
Comment (MaoShizhong) [inline ruby_on_rails/mailers_advanced_topics/mailers.md:59]:
```suggestion
```

File: ruby_on_rails/rails_sprinkles/stimulus_js.md [hunk 3]
```diff
@@ -296,18 +298,18 @@ button.
 count (imagine a user writing a post which has a maximum length of 280 characters)
 - **Project**:
 Go back to you **Flight Booker** project and improve it:
-  - Add a controller that allows the user to add another passenger by clicking on an "Add passenger" button, which adds another set of fields to enter the passenger details (hint: have a look at the [\<template\>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template) tag)
+  - Add a controller that allows the user to add another passenger by clicking on an "Add passenger" button, which adds another set of fields to enter the passenger details (hint: have a look at the [```html<template>``` tag](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template) ).
```
Comment (MaoShizhong) [inline ruby_on_rails/rails_sprinkles/stimulus_js.md:301]:
```suggestion
  - Add a controller that allows the user to add another passenger by clicking on an "Add passenger" button, which adds another set of fields to enter the passenger details (hint: have a look at the [\<template\> tag](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template)).
```

**Review by MaoShizhong — CHANGES_REQUESTED** [review]
Regarding the link fragment, looks like a difference in algorithm for heading conversion between Rails and markdownlint when handling the `/`. We'll explore that separately.

For.now, you can change the fragment to just `#selecting-elements`, then change the relevant heading to `Selecting elements` i.e. get rid of the `/ targeting` part, as it's not actually needed for the heading.

**Comment by thatsthat** [comment]
Ok, I think I've addressed all the issues you pointed out.

**Review by MaoShizhong — APPROVED** [review]
:pray:

**Pull request merged**